tools/src/main/java/it/eng/tools/daps/DapsProperties.java

```diff
@@ -13,24 +16,138 @@
+import it.eng.tools.service.ApplicationPropertiesService;
+//import it.eng.tools.service.ApplicationPropertiesService;
+//import lombok.Data;
+//import lombok.NoArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 
-@ConfigurationProperties(prefix = "application.daps")
-@Configuration
-@Data
-@NoArgsConstructor
+//@ConfigurationProperties(prefix = "application.daps")
+//@Configuration
+//@Data
+//@NoArgsConstructor
+@Component
 @Slf4j
 public class DapsProperties {
 
-	private boolean enabledDapsInteraction;
+	private final String PREFIX = "application.daps.";
+	private final String ENABLED_DAPS_INTERACTION_KEY = PREFIX + "enabledDapsInteraction";
+	private final String EXTENDED_TOKEN_VALIDATION_KEY = PREFIX + "extendedTokenValidation";
+	private final String TOKEN_CACHING_KEY = PREFIX + "tokenCaching";
+	private final String FETCH_TOKEN_ON_STARTUP_KEY = PREFIX + "fetchTokenOnStartup";
+	private final String DAPS_URL_KEY = PREFIX + "dapsUrl";
+	private final String DAPS_JWKS_URL_KEY = PREFIX + "dapsJWKSUrl";
+	private final String DAPS_KEYSTORE_NAME_KEY = PREFIX + "dapsKeyStoreName";
+	private final String DAPS_KEYSTORE_PASSWORD_KEY = PREFIX + "dapsKeyStorePassword";
+	private final String DAPS_KEYSTORE_ALIAS_NAME_KEY = PREFIX + "dapsKeystoreAliasName";
+
+	/*private boolean enabledDapsInteraction;
 	private boolean extendedTokenValidation;
 	private boolean tokenCaching;
 	private boolean fetchTokenOnStartup;
 	private URL dapsUrl;
-	private URL dapsJWKSUrl;
-	
+	private URL dapsJWKSUrl; */
+
+	private final ApplicationPropertiesService service;
+
+	public DapsProperties(ApplicationPropertiesService service) { 
+		this.service = service;
+	}
+
+
+	protected boolean isEnabledDapsInteraction() { 
+		String enabledDapsInteractionValue = service.get(ENABLED_DAPS_INTERACTION_KEY);
+
+		if(enabledDapsInteractionValue != null) {
+			return Boolean.valueOf(enabledDapsInteractionValue);
+		}
+		return false;
+	}
+
+	protected boolean isExtendedTokenValidation() {
+		String extendedTokenValidationValue = service.get(EXTENDED_TOKEN_VALIDATION_KEY);
+
+		if(extendedTokenValidationValue != null) {
+			return Boolean.valueOf(extendedTokenValidationValue);
+		}
+		return false;
+	}
+
+	protected boolean isTokenCaching() {
+		String tokenCachingValue = service.get(TOKEN_CACHING_KEY);
+
+		if(tokenCachingValue != null) {
+			return Boolean.valueOf(tokenCachingValue);
+		}
+		return false;
+	}
+
+	protected boolean isFetchTokenOnStartup() {
+		String fetchTokenOnStartupValue = service.get(FETCH_TOKEN_ON_STARTUP_KEY);
+
+		if(fetchTokenOnStartupValue != null) {
+			return Boolean.valueOf(fetchTokenOnStartupValue);
+		}
+		return true;
+	}
+
+	protected URL getDapsUrl() {
+		String dapsUrlValue = service.get(DAPS_URL_KEY);
+
+		URL dapsUrl = null;
+		try {
+			dapsUrl = new URL(dapsUrlValue);
+		} catch (MalformedURLException e) {
+			throw new DapsPropertyErrorException(DAPS_URL_KEY + " isn't defined!!!");
```

Comment:
Could it be also invalid URL value?

Reply:
Do we have this service covered with junit tests?
if value is present, not present and has invalid URL value, new URL("abc")
(do not know if last one will work or throw new MalformedUrlException)

Reply:
There isn't any specific test. I could create a new JUnit class or add these tests in DapsServiceOmejdnTest.java or DapsCacheTest.java (which already reference the class DapsProperties).